__tests__/app.test.js

```diff
@@ -56,47 +56,81 @@ describe("/api", () => {
+                                body: 'I find this existence challenging',
+                                created_at: '2020-07-09T20:11:00.000Z',
+                                votes: 100,
+                                article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+                            })
                         })
                     })
                 })
-            })
 
-        describe("Error handling", () => {
-            test("returns status code 400 when given an invalid article_id", () => {
-                return request(app)
-                .get("/api/articles/number-one-as-a-string")
-                .expect(400)
-                .then(({body}) => {
-                    expect(body.msg).toBe("bad request")
+            describe("Error handling", () => {
+                test("returns status code 400 when given an invalid article_id", () => {
+                    return request(app)
+                    .get("/api/articles/number-one-as-a-string")
+                    .expect(400)
+                    .then(({body}) => {
+                        expect(body.msg).toBe("bad request")
+                    })
                 })
-            })
-                    
-            test("returns status code 404 when given an article_id of the right data type that doesnt exist", () => {
-                return request(app)
-                .get("/api/articles/99")
-                .expect(404)
-                .then(({body}) => {
-                    expect(body.msg).toBe("no article found with an article_id of 99")
+                        
+                test("returns status code 404 when given an article_id of the right data type that doesnt exist", () => {
+                    return request(app)
+                    .get("/api/articles/99")
+                    .expect(404)
+                    .then(({body}) => {
+                        expect(body.msg).toBe("no article found with an article_id of 99")
+                    })
                 })
             })
         })
-    })
-})
\ No newline at end of file
+
+        /*
+        Tests:
+            - responds with an articles array of article objects containing all of the correct values: author, title, article_id, topic, created_at, votes, article_img_url, comment_count"
+        */
+
+        describe("/api/articles", () => {
+            describe("Status 200", () => {
+                test("responds with an articles array of article objects containing all of the correct values: author, title, article_id, topic, created_at, votes, article_img_url, comment_count", () => {
+                    return request(app)
+                    .get("/api/articles")
+                    .expect(200)
+                    .then(({body}) => {
+                        const responseData = body.articles
+                            responseData.forEach((article) => {
```

Comment:
When we do a forEach we need to make sure we have some articles, check the test data and count how many there should be, and expect the length. If we don't do this there could be 0 and if there was 0 the callback function inside the forEach would never get called and thus the expect statements would never happen